xonsh/readline_shell.py

```diff
@@ -230,6 +231,14 @@ def __init__(self, completekey='tab', stdin=None, stdout=None, **kwargs):
     def __del__(self):
         teardown_readline()
 
+
+    def sttysane(self):
+        """Change settings of stdin to include ECHO so characters typed at the keyboard are echoed on screen"""
+        fd = sys.stdin.fileno()
+        si_attrs = termios.tcgetattr(fd)
+        si_attrs[3] = si_attrs[3] & termios.ECHO
+        termios.tcsetattr(fd, termios.TCSADRAIN, si_attrs)
```

Comment:
Yeah, not sure why this doesn't work, but maybe it has to do with readline taking over stdin.